app/models/audio_file.rb

```diff
@@ -94,4 +99,21 @@ def set_status
       self.status_message = errors
     end
   end
+
+  def analyze!
+    return if status != UPLOADED
+
+    submit_porter_job "cms::audio_file:analyze:#{id}", asset_url do
```

Comment:
Ooh, interesting.  This reminds me of rails Global Ids ... any reason not to use those?  I guess they don't include any notion of which job they were for (`analyze`), unless we extended them in some way.  But at the moment, it's only 1 job-type per model right?

```ruby
> UserImage.last.to_global_id.to_s
=> "gid://prx/UserImage/43920"
> GlobalID::Locator.locate 'gid://prx/UserImage/43920'
=> #<UserImage id: 43920, ...>
```

And crazy idea: move images over to GIDs and we can drop that porter_job_id field?  (It's easier to get rid of before this makes it to prod).

Reply:
Though I do wonder if this could cause problems with order-of-callbacks.

Reply:
Update: I don't think out-of-order callbacks will be an issue in real life.  These `process!` methods are only called by the controllers, and if we need to debounce a bit could just add a `def process!(force = false)` flag, that normally skips processing if status == UPLOADED.  (Similar to how feeder does this).